annotations/tekton-annotations/src/main/java/io/dekorate/tekton/manifest/TektonManifestGenerator.java

```diff
@@ -389,6 +385,10 @@ public void generatePipelineResources(TektonConfig config) {
       resourceRegistry.decorate(TEKTON_PIPELINE_RUN,
           new AddPvcToPipelineRunDecorator(null, PIPELINE_M2_WS, m2WorkspaceClaimName, false));
     }
+
+    resourceRegistry.decorate(TEKTON_PIPELINE, new AddServiceAccountResourceDecorator(imageBuildTaskName));
```

Comment:
I think this is not correct. Before, this decorator was using `config.getName()` and then it will retrieve the `Deployment` (or Service, DeploymentConfig, ...) resources by this name (see line https://github.com/dekorateio/dekorate/pull/853/files#diff-11fbc30c8c05acda88e00bd0e05f40d2dd1fda355ca6b672e2918a1354c78b05R37).

However, this decorator now uses `imageBuildTaskName`, so the Deployment resources will never be found.

Reply:
Before, this argument was just used to name the resource and didn't affect discovery of existing resources.
Now, that discovery by name is introduced we need to make sure that we pass the correct name, in this case `imageBuildTaskName`.

The discovery part works fine, according to the tests, but you may be right that this change in naming may introduce hiccups. I'll have check to see if we have issues there.

Reply:
+1